src/NEPTypes.jl

```diff
@@ -228,11 +252,23 @@ julia> compute_Mder(nep,1)-(A0+A1*exp(1))
     end
     function compute_Mder(nep::SPMF_NEP,λ::Number,i::Integer=0)
         if (i==0)
-            Z=copy(nep.Zero)
-            for i=1:size(nep.A,1)
-                Z=Z+nep.A[i]*nep.fi[i](reshape([λ],1,1))[1]
+            if isempty(nep.As)
+                Z=copy(nep.Zero)
+                for i=1:size(nep.A,1)
+                    Z=Z+nep.A[i]*nep.fi[i](reshape([λ],1,1))[1]
+                end
+                return Z
+            else
+                T = Complex{Float64}
+                x = T(nep.fi[1](reshape([λ],1,1))[1])
+                Z = copy(nep.As[1])
+                Z.nzval .*= x
+                @inbounds for a = 2:length(nep.As)
+                    x = T(nep.fi[a](reshape([λ],1,1))[1])
+                    Z.nzval .+= nep.As[a].nzval .* x
```

Comment:
The way to handle types here is a bit tricky. I would say `T` should be determined from the "greatest" type of `eltype(nep.Av[i])` and `typeof(nep.fi[i](reshape([λ],1,1))`, e.g., if one of them is `Complex{Float64}` but all others are `Float64` then `T=Complex{Float64}`.  I think "greatest" can be determined with [the `promote_rule` commands](https://docs.julialang.org/en/v0.6.2/manual/conversion-and-promotion/). We would maybe first need to compute a vector `Vector{Any}` of all the fi-values.

Reply:
Should be fixed now, without loss of performance it seems.